src/form.rs

```diff
@@ -42,6 +42,8 @@ use serde::Serialize;
 pub trait Form {
 	/// The type identifier type.
 	type TypeId: PartialEq + Eq + PartialOrd + Ord + Clone + core::fmt::Debug;
+	/// The string type.
+	type String: Serialize + PartialEq + Eq + PartialOrd + Ord + Clone + core::fmt::Debug;
```

Comment:
While I like the overall idea I think it is better in this case to opt-into KISS (keep it simple stupid) and instead convert all `&'static str` simply into `String`. The `&'static str` was used in the beginning when we did not need deserialization in ink!. Now that we want to support Substrate we need it. However, if we do not generate metadata in any performance critical code section we don't need to opt-in for maximum performance and can instead opt-into simple code.

Reply:
I've removed the `OwnedForm` now, but kept `MetaForm` as `& 'static str`, and changed `CompactForm` to `String`. Let me know what you think.